fastlane/lib/fastlane/plugins/plugin_info_collector.rb

```diff
@@ -0,0 +1,67 @@
+module Fastlane
+  class PluginInfoCollector
+    def initialize(ui = PluginGeneratorUI.new)
+      @ui = ui
+    end
+
+    def collect_info
+      plugin_name = collect_plugin_name
+      author = collect_author
+
+      PluginInfo.new(plugin_name, author)
+    end
+
+    #
+    # Plugin name
+    #
+
+    def collect_plugin_name
+      plugin_name = nil
+      loop do
+        plugin_name = @ui.input("What would you like to be the name of your plugin?")
```

Comment:
This method will detect if the user types in the prefix. The idea is that `plugin_name_valid?` will find the prefix and return falsey, which causes `fix_plugin_name` to run and produce a name suggestion that has the prefix removed. 👍